### DISCUSSION TITLE
Run legendary on wineskin?

### DISCUSSION THREAD

**Discussion by bearcatjamboree** [legendary-gl/legendary#414] [General]
Is it possible to run games on Mac without crossover?  For instance, can i just create a bottle called Legendary.app and use it to run games through legendary-gl instead of using crossover?

**Comment by derrod** [comment]
Probably, I just don't know how since I've never used wineskin. But you can use the manual options for the wine binaries etc. for that.

**Reply by cboxgo69** [reply]
Thanks I’ll try the manual options

**Reply by bearcatjamboree** [reply]
Quick question:  Is there a way to use something like "disable_auto_crossover = true" to use wine and not use crossover?  It looks like this flag has no effect on cli.py and you still get the following error trying to use just wine64: 

[cli] ERROR: Attempted to use default CrossOver bottle ("Legendary"), but it does not exist, see https://legendary.gl/crossover-setup for setup instructions.

**Comment by bearcatjamboree** [comment]
I think I might have found the issue with disable_auto_crossover not working.  Core.py has the following around line 589:

```
        # override wine prefix if necessary
        if sys_platform == 'darwin':
            if cx_bottle:
                env['CX_BOTTLE'] = cx_bottle
            elif 'CX_BOTTLE' not in os.environ:  # Only fall back to config is not already set in env
                cx_bottle = self.lgd.config.get('default', 'crossover_bottle', fallback='Legendary')
                cx_bottle = self.lgd.config.get(app_name, 'crossover_bottle', fallback=cx_bottle)
                if cx_bottle:
                    env['CX_BOTTLE'] = cx_bottle
            else:
                cx_bottle = os.environ['CX_BOTTLE']
            self.log.info(f'Using CrossOver Bottle "{cx_bottle}"')

```
But, maybe it should be fallback=None (instead of Legendary) to keep from adding cx_bottle to the environment which will cause it to think that crossover was selected later in the code.  Here is what I mean:

```
        # override wine prefix if necessary
        if sys_platform == 'darwin':
            if cx_bottle:
                env['CX_BOTTLE'] = cx_bottle
            elif 'CX_BOTTLE' not in os.environ:  # Only fall back to config is not already set in env
                cx_bottle = self.lgd.config.get('default', 'crossover_bottle', fallback=None)
                cx_bottle = self.lgd.config.get(app_name, 'crossover_bottle', fallback=cx_bottle)
                if cx_bottle:
                    env['CX_BOTTLE'] = cx_bottle
            else:
                cx_bottle = os.environ['CX_BOTTLE']
            self.log.info(f'Using CrossOver Bottle "{cx_bottle}"')

```

This causes the condition at line #647 to get triggered in cli.py producing the following message:

"Attempted to use default CrossOver bottle ("Legendary"), but it does not exist, see https://legendary.gl/crossover-setup for setup instructions."

**Reply by derrod** [reply]
Ah yeah, probably gotta fix that with the next release (this month, I swear).

**Comment by bearcatjamboree** [comment]
another question:

When you launch an app, do you sometimes have it launch a separate exe after the first returns a value?  I wonder because Fortnite runs a separate exe on startup, but the path that you have stored in installed.json appears to get dropped so it doesn't know where the 2nd exe is located.

**Reply by derrod** [reply]
Some games have launchers that then run the actual game (e.g. due to anticheat), Legendary doesn't handle any of that, it just runs the exe specified in the manifest, everything else is left up to the game.

**Comment by bearcatjamboree** [comment]
Is it possible to set the WINEPATH value to the path of the EXE that you call?  or is there a way to include a helper script of some kind? 
 In other words, does the config.ini support having a script that you pass the command to, so maybe it could parse the path and setup the WINEPATH?

**Comment by bearcatjamboree** [comment]
Actually, I think I might have found a way using the AppName.env..
